pkgs/applications/science/logic/avy/default.nix

```diff
@@ -1,13 +1,13 @@
 { stdenv, fetchgit, cmake, zlib, boost }:
 
-stdenv.mkDerivation {
+stdenv.mkDerivation rec {
   pname = "avy";
-  version = "2017.10.16";
+  version = "cav19";
```

Comment:
The version must start with a decimal digit, this will be interpreted as having package name "avy-cav19":

    nix-repl> builtins.parseDrvName "avy-cav19"
    { name = "avy-cav19"; version = ""; }

I would suggest continuing with the date based format and put cav19 as a comment, i.e., something like

    version = "2020-mm-dd"; # Date of the "cav19" tag.

Reply:
Thanks for catching that!
You can suggest a code change with the "Insert suggestion" button or hitting `ctrl+g` by the way. That way I can immediately apply the suggestion from the GitHub interface.

Reply:
The problem with that way is it leaves us with loads of tiny commits and includes intermediate non-working versions. I find if I'm going to pull that back down and do a squash it's usually just as much work to make the change myself.

Reply:
And you know what I'm going to ask now ;)

Could you squash your fix in? It just leaves an unnecessarily confusing history.

Reply:
I was specifically keeping it as a separate commit, to make it easier to see that I've addressed your comment.
Then whoever merges this can just choose "squash & merge".

Reply:
:grimacing: squash & merge leads to the opposite problem, losing possibly important detail (and bisect-ability) between logically separable changes.

Reply:
But not for something this small, where my manual squash would end up with a single commit anyway. I'm all for logically structured commits for larger changes.
But my preferred workflow is usually:
- initial version of PR
- address comments in individual commits
- rebase/squash right before merging (if multiple commits are needed do it manually, if one is sufficient, just use the GitHub UI)

Though I'm not going to argue much over such detail. I've squashed & pushed.